File: smells/cliargs.js
const vscode = require('vscode');

 const smell = {

    detect : (token) => {

        if(token.hasOwnProperty("line")) var lineno = token.line;
        if(token.hasOwnProperty("type")) var tokenType = token.type;
        if(token.hasOwnProperty("name")) var name= token.name;
        if(token.hasOwnProperty("args")) var args= token.args;
        if(token.hasOwnProperty("hasInputs")) var hasInputs= token.hasInputs;
        
        const cliArgsFuncNames = ['sys.argv','ArgumentParser','argparse','subprocess.Popen'];

        if(tokenType=="function_call" && cliArgsFuncNames.includes(name) && hasInputs == "True"){

            const warning = 'possible use of command line args at line '+ lineno;
            vscode.window.showWarningMessage(warning);
        }
    }
}

module.exports = smell;

File: py/smells/cliargs.py
from operations.savewarnings import saveWarnings

def detect(token):

    if token.__contains__("line"): lineno = token["line"] 
    if token.__contains__("type"): tokenType = token["type"]
    if token.__contains__("name"): methodname = token["name"]
    if token.__contains__("args"): args = token["args"]
    if token.__contains__("hasInputs"): containsUserInput =  token["hasInputs"]

    cliArgsFuncNames = ['sys.argv', 'ArgumentParser', 'argparse', 'subprocess.Popen']
    
    if tokenType == "function_call" and methodname in cliArgsFuncNames and containsUserInput:
        
        warning = 'use of command line args' 
        saveWarnings(warning,str(lineno))
        print(warning+ ' at line '+ str(lineno))
